app/services/project_job_service.rb

```diff
@@ -6,6 +6,7 @@ def initialize(project:)
   end
 
   def list_contents_job(user:)
-    ListProjectContentsJob.perform_now(user_id: user.id, project_title: @project.title)
+    job = ListProjectContentsJob.perform_later(user_id: user.id, project_id: @project.id)
+    UserJob.create_and_link_to_user(job_id: job.job_id, user: user, job_title: "File list for #{@project.title}")
```

Comment:
This makes sure the `UserJob` record is created as soon as the job is queued (regardless of how long it takes to be picked up) and allows us to show it to the user as in progress.